### PR TITLE
Feat/edit mapping

### PR DESCRIPTION
**Pull request by akshay-codemonk**
Add jQuery functions to handle editing of mapping values on click of a mapping table row

screenshot:
edit_mapping

### CODE REVIEW

File: apps/task/tasks.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
+from django_q.tasks import async_task, Chain
+
+from apps.expense.models import Expense, ExpenseGroup
+from apps.task.models import TaskLog
+from apps.xero_workspace.models import Workspace, Invoice, InvoiceLineItem
+from apps.xero_workspace.utils import connect_to_fyle, connect_to_xero
+
+
+def create_task_log(workspace_id, expense_group_id=None, chain=None):
```
Comment (snarayanank2) [inline apps/task/tasks.py]: I am not grokking this part

File: apps/task/tasks.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    """
+    Run async tasks and create task log object
+    :param workspace_id
+    :param expense_group_id
+    :param chain
+    :return
+    """
+
+    if expense_group_id is None:
```
Comment (snarayanank2) [inline apps/task/tasks.py]: this function seems problematic.. it is doing very different things depending on parameter - can we refactor this to be more obvious

File: apps/task/tasks.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+
+def sync_to_xero(expense_group_id):
+    """
+    Create invoice, invoice line items and post to Xero
+    :param expense_group_id
+    :return: expense_group_id
+    """
+
+    expense_group = ExpenseGroup.objects.get(id=expense_group_id)
```
Comment (snarayanank2) [inline apps/task/tasks.py]: this logic should be in expense group model. the task should simply call the model method

File: apps/task/tasks.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    invoice_id = Invoice.create_invoice(expense_group)
+    InvoiceLineItem.create_invoice_line_item(invoice_id, expense_group)
+    xero = connect_to_xero(expense_group.workspace.id)
+    invoice_data = generate_invoice_request_data(invoice_id)
+    post_to_xero(invoice_data, xero)
+    return expense_group_id
+
+
+def generate_invoice_request_data(invoice_id):
+    """
```
Comment (snarayanank2) [inline apps/task/tasks.py]: all of this logic should be in the invoice model

File: apps/task/tasks.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+
+def post_to_xero(data, xero):
+    """ Makes an API call to create invoices in Xero
+    :param data: Request data for the invoice API
+    :param xero: Xero connection object
+    :return response: response data from Xero API
+    """
+
+    response = xero.invoices.post(data)
```
Comment (snarayanank2) [inline apps/task/tasks.py]: belongs to the invoice model

File: apps/task/tasks.py (hunk 2 above)
Comment (akshay-codemonk) [reply apps/task/tasks.py]: ok

**Pull request merged**